opentelemetry-otlp/tests/integration_test/src/test_utils.rs

```diff
@@ -0,0 +1,145 @@
+//! Supporting infrastructure for OTLP integration tests.
+//!
+//! This module provides the pieces needed to work with an actual opentelemetry-collector
+//! instance, which is started in Docker and has its output plumbed back into the host filesystem.
+//! This lets us write tests that push data over OTLP (HTTP or gRPC) to the collector, and then read
+//! that data back from the filesystem to ensure everything worked out as expected.
+//!
+//! To use this module, all you need to do is call `start_collector_container()` from each
+//! of your tests, and use a single `#[dtor]` at the end of your test file to call
+//! `stop_collector_container`. Note that as cargo integration tests run a process-per-test-file,
+//! each test will get its own fresh instance of the container.
+//!
+//! Only a single test suite can run at once, as each container has statically mapped ports, but
+//! this works nicely with the way cargo executes the suite.
+//!
+//! To skip integration tests with cargo, you can run `cargo test --mod`, which will run unit tests
+//! only.
+//!
+#![cfg(unix)]
+
+use anyhow::Result;
+use opentelemetry::{otel_debug, otel_info};
+use std::fs;
+use std::fs::File;
+use std::os::unix::fs::PermissionsExt;
+use std::sync::{Arc, Mutex, Once, OnceLock};
+use testcontainers::core::wait::HttpWaitStrategy;
+use testcontainers::core::{ContainerPort, Mount};
+use testcontainers::{core::WaitFor, runners::AsyncRunner, ContainerAsync, GenericImage, ImageExt};
+use tracing_subscriber::FmtSubscriber;
+
+// Static references for container management
+static COLLECTOR_ARC: OnceLock<Mutex<Option<Arc<ContainerAsync<GenericImage>>>>> = OnceLock::new();
+
+pub static METRICS_FILE: &str = "./actual/metrics.json";
+pub static LOGS_FILE: &str = "./actual/logs.json";
+pub static TRACES_FILE: &str = "./actual/traces.json";
+
+static INIT_TRACING: Once = Once::new();
+
+fn init_tracing() {
+    INIT_TRACING.call_once(|| {
+        let subscriber = FmtSubscriber::builder()
+            .with_max_level(tracing::Level::DEBUG)
```

Comment:
DEBUG is giving a ton of noise from the networking libraries in the CI logs.. probably okay for now, we can revisit if we find it too much noise.